ui/apps/platform/src/Containers/NetworkGraph/simulation/NetworkPoliciesGenerationScope.tsx

```diff
@@ -25,54 +26,74 @@ function NetworkPoliciesGenerationScope({
         const namespaceCount = namespaces.length;
         const namespaceText =
             namespaceCount === 1 ? `namespace "${namespaces[0]}"` : `${namespaceCount} namespaces`;
         namespaceElement = <span>{namespaceText}</span>;
-    }
 
-    if (granularity === 'DEPLOYMENT') {
         const deploymentCount = deployments.length;
         const deploymentText =
             deploymentCount === 1
                 ? `deployment "${deployments[0].name}"`
                 : `${deploymentCount} deployments`;
 
-        deploymentElement = <span>{deploymentText}</span>;
+        deploymentElement = (
+            <span>
+                <Button
+                    variant="link"
+                    isInline
+                    onClick={() => setModalDeployments(networkPolicyGenerationScope.deployments)}
+                >
+                    {deploymentText}
+                </Button>
+            </span>
+        );
     }
 
     return (
-        <div className="network-policies-generation-scope">
-            <Flex
-                alignItems={{ default: 'alignItemsCenter' }}
-                spaceItems={{ default: 'spaceItemsSm' }}
-            >
-                <DeploymentIcon />
-                {deploymentElement}
-            </Flex>
+        <>
+            {modalDeployments && (
+                <DeploymentScopeModal
+                    deployments={modalDeployments}
+                    isOpen={modalDeployments !== null}
+                    onClose={() => setModalDeployments(null)}
+                />
+            )}
+            <div className="network-policies-generation-scope">
+                <Flex
+                    alignItems={{ default: 'alignItemsCenter' }}
+                    spaceItems={{ default: 'spaceItemsSm' }}
+                >
+                    <DeploymentIcon />
+                    {deploymentElement}
+                </Flex>
 
-            <Flex
-                alignItems={{ default: 'alignItemsCenter' }}
-                spaceItems={{ default: 'spaceItemsSm' }}
-            >
-                <NamespaceIcon />
-                {namespaceElement}
-            </Flex>
+                <Flex
+                    alignItems={{ default: 'alignItemsCenter' }}
+                    spaceItems={{ default: 'spaceItemsSm' }}
+                >
+                    <NamespaceIcon />
+                    {namespaceElement}
+                </Flex>
 
-            <Flex
-                alignItems={{ default: 'alignItemsCenter' }}
-                spaceItems={{ default: 'spaceItemsSm' }}
-            >
-                <ClusterIcon />
-                <span>cluster &quot;{cluster}&quot;</span>
-            </Flex>
-        </div>
+                <Flex
+                    alignItems={{ default: 'alignItemsCenter' }}
+                    spaceItems={{ default: 'spaceItemsSm' }}
+                >
+                    <ClusterIcon />
+                    <span>cluster &quot;{cluster}&quot;</span>
```

Comment:
Something seems off in the communication when there is CL badge/icon and word cluster "whatever"

One suggestion is to replace quote marks with key-value-ish presentation CL cluster: whatever